src/gui/tray/ActivityListModel.cpp

```diff
@@ -461,6 +465,28 @@ void ActivityListModel::triggerActionAtIndex(int id) const
     }
 }
 
+void ActivityListModel::handleAction(int activityIndex, int actionIndex)
+{
+    if (activityIndex < 0 || activityIndex >= _finalList.size()) {
+        qCWarning(lcActivity) << "Couldn't handle action on activity at index" << activityIndex << "/ final list size:" << _finalList.size();
+        return;
+    }
+
+    const auto actions = _finalList[activityIndex]._links;
+
+    if (actionIndex < 0 || actionIndex >= actions.size()) {
+        qCWarning(lcActivity) << "Couldn't handle action at index" << actionIndex << "/ actions list size:" << actions.size();
```

Comment:
As far as logging is concerned, it might be useful to also mention the activityIndex there.